src/core/algorithms/dc/model/dc.cpp

```diff
@@ -0,0 +1,104 @@
+#include "algorithms/dc/model/dc.h"
+
+#include <algorithm>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "algorithms/dc/model/operator.h"
+#include "algorithms/dc/model/predicate.h"
+
+namespace algos::dc {
+
+bool DC::CheckAllEquality() const {
+    auto check = [](auto const& pred) {
+        return pred.GetOperator() == dc::OperatorType::kEqual and pred.IsCrossTuple() and
+               pred.IsOneColumn();
+    };
+
+    return std::all_of(predicates_.begin(), predicates_.end(), check);
```

Comment:
you can ignore it:
`std::ranges` allows to get rid of boilerplate:
```suggestion
    return std::ranges::all_of(predicates_, check);
```